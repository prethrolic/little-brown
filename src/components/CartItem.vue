<template>
  <div class='cit-container'>
    <img class='cit-cover-img' :src='book.cover'>
    <div v-html='book.title' class='cit-book-title' />
    <div class='cit-count-modifier'>
      <div class='cit-button' @click='onMinusClick'>
        <div class='icon icon-minus' />
      </div>
      <div v-html='count' class='cit-count-display' />
      <div class='cit-button' @click='onPlusClick'>
        <div class='icon icon-plus' />
      </div>
    </div>
    <div v-html="'à¸¿' + totalPrice" class='cit-total-price' />
  </div>
</template>
<script>
export default {
  name: 'CartItem',
  props: {
    book: Object,
    count: Number,
  },
  methods: {
    onPlusClick() {
      this.$store.dispatch('add_to_cart', this.book);
    },
    onMinusClick() {
      this.$store.dispatch('remove_from_cart', this.book);
    },
  },
  computed: {
    totalPrice() {
      return this.book.price * this.count;
    },
  },
};
</script>
<style lang="scss" scoped>
@import '@/assets/scss/variables.scss';
@import '@/assets/scss/icons.scss';

.cit {
  &-container {
    display: grid;
    align-items: center;
    grid-template-columns: .8fr 2fr 2fr 1fr;
    width: 100%;
    grid-gap: 16px;
    margin-bottom: 16px;
  }

  &-cover-img {
    max-height: 100%;
    max-width: 100%;
    border-radius: 8px;
    display: block;
  }

  &-count-modifier {
    display: flex;
    align-items: center;
  }

  &-button {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 32px;
    height: 32px;
    border: none;
    background-color: $light-gray;
    border-radius: 8px;
    cursor: pointer;

    &:hover {
      background-color: $light-gray-hover;
      transition: 0.2s ease-in-out;
    }
  }

  &-count-display {
    flex-grow: 1;
    font-size: 14px;
    font-weight: 600;
  }

  &-total-price {
    font-size: 14px;
    font-weight: 600;
    text-align: right;

  }

  &-book-title {
    text-align: left;
    font-size: 12px;
    max-height: 48px;
    font-weight: 600;
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    overflow-y: hidden;
  }
}

@media(max-width: 600px) {
  .cit {
    &-container {
      display: grid;
      align-items: center;
      grid-template-columns: 2.5fr 2fr 1fr;
      width: 100%;
      grid-gap: 16px;
      margin-bottom: 16px;
    }

    &-cover-img {
      display: none;
    }
    &-total-price, &-count-display {
      font-size: 10px;
    }

    &-book-title {
      font-size: 10px;
      -webkit-line-clamp: 2;
    }
  }
}
</style>
