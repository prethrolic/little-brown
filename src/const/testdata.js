export const bookList = [
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg",
    "price": "350",
    "title": "Harry Potter and the Philosopher's Stone (I)",
    "id": "9781408855652"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855669.jpg",
    "price": "350",
    "title": "Harry Potter and the Chamber of Secrets (II)",
    "id": "9781408855669"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855676.jpg",
    "price": "340",
    "title": "Harry Potter and the Prisoner of Azkaban (III)",
    "id": "9781408855676"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9780/2419/9780241975367.jpg",
    "price": "230",
    "title": "The Monk of Mokha",
    "id": "9780241975367"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/7847/9781784706883.jpg",
    "price": "300",
    "title": "The Language of Kindness",
    "id": "9781784706883"
  },
];

export const cart = {
  "9781408855652": {
    book: {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg",
    "price": "350",
    "title": "Harry Potter and the Philosopher's Stone (I)",
    "id": "9781408855652"
    },
    count: 2,
  },
  "9781408855669": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855669.jpg",
      "price": "350",
      "title": "Harry Potter and the Chamber of Secrets (II)",
      "id": "9781408855669"
    },
    count: 1,
  },
  "9781408855676": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855676.jpg",
      "price": "340",
      "title": "Harry Potter and the Prisoner of Azkaban (III)",
      "id": "9781408855676"
    },
    count: 1,
  },
  "9780241975367": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9780/2419/9780241975367.jpg",
      "price": "230",
      "title": "The Monk of Mokha",
      "id": "9780241975367"
    },
    count: 2,
  },
};

export const inPromotion = ["9781408855652", "9781408855669", "9781408855676"];

export const user = {
  displayName: 'John Doe',
  pictureUrl: 'https://pbs.twimg.com/profile_images/1269585418107805697/6QKUkoWJ_400x400.jpg',
};

export const paymentInfo = {
  subtotal: '1850.00',
  discount: '114.40',
  total: '1735.60',
  cash: '2000',
};