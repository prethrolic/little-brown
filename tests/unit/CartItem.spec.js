import { shallowMount } from '@vue/test-utils';
import CartItem from '@/components/CartItem.vue';
import { bookList } from '@/const/testdata.js';

describe('CartItem.vue', () => {
  it('render cart items correctly', () => {
    const wrapper = shallowMount(CartItem, {
      propsData: {
        book: bookList[0],
        count: 2,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});