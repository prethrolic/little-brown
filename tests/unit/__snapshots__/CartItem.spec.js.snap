// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem.vue render cart items correctly 1`] = `
<div
  class="cit-container"
>
  <img
    class="cit-cover-img"
    src="https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg"
  />
   
  <div
    class="cit-book-title"
  >
    Harry Potter and the Philosopher's Stone (I)
  </div>
   
  <div
    class="cit-count-modifier"
  >
    <div
      class="cit-button"
    >
      <div
        class="icon icon-minus"
      />
    </div>
     
    <div
      class="cit-count-display"
    >
      2
    </div>
     
    <div
      class="cit-button"
    >
      <div
        class="icon icon-plus"
      />
    </div>
  </div>
   
  <div
    class="cit-total-price"
  >
    à¸¿700
  </div>
</div>
`;
